### PR TITLE
Live times

### PR DESCRIPTION
**Pull request by f11xter**

### CODE REVIEW

File: src/models/journey.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+export type journey = {
```
Comment (WendiF) [inline src/models/journey.ts:1]: types and type files should be capitalized please

File: src/models/station.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+export type stationAPI = {
+    aliases: any[];
+    crs: string;
+    id: number;
+    isGroupStation: boolean;
+    isSilverSeekStation: boolean;
+    latitude: number;
+    longitude: number;
+    name: string;
+    nlc: string;
```
Comment (WendiF) [inline src/models/station.ts:15]: nitpick - is this the order which the respoonse gives the info? It's a bit of a weird order.

Also, don't use type: `any` where possible (definitely possible here). Probably a `string[]` would work?

File: src/helpers/ApiCallHelper.tsx [hunk 3]
```diff
@@ -1,7 +1,25 @@
...
+        .then((data) => data.stations);
+};
+
+export const fetchFares = (origin: string, destination: string, datetime: string, children = '0', adults = '1') => {
+    return fetch('https://mobile-api-softwire1.lner.co.uk/v1/fares?' + new URLSearchParams({
+        originStation: origin,
+        destinationStation: destination,
+        outboundDateTime: datetime,
+        numberOfChildren: children,
+        numberOfAdults: adults,
```
Comment (WendiF) [inline src/helpers/ApiCallHelper.tsx:17]: nitpick - I'd prefer to extract the urlSearchParams out and then use template literals to form the url inside the `fetch` - just so it's a bit more readable

File: src/components/Stations.tsx [hunk 4]
```diff
@@ -1,45 +1,96 @@
...
+                                <td>{journey.departureTime.toLocaleTimeString()}</td>
+                                <td>{journey.origin}</td>
+                                <td>{journey.destination}</td>
+                                <td>{journey.arrivalTime.toLocaleTimeString()}</td>
+                            </tr>)}
+                    </tbody>
+                </table>
+            </div>
+
+            }
```
Comment (WendiF) [inline src/components/Stations.tsx:93]: This definitely can be extracted into a component (I'm assuming you did do this in a later PR but noting it here anyway

File: src/components/Stations.tsx [hunk 5]
```diff
@@ -1,45 +1,96 @@
...
+    today.setHours(today.getHours() + 1);
+
+    const [isLoading, setLoading] = useState(false);
 
     const handleClick = () => {
-        window.location.href = `https://www.lner.co.uk/travel-information/travelling-now/live-train-times/depart/${from}/${to}/#LiveDepResults`;
+        setLoading(true);
+        setJourneys(undefined);
+        fetchFares(from, to, today.toISOString())
+            .then((value) => setJourneys(value.map((journey: any): journey => ({
```
Comment (WendiF) [inline src/components/Stations.tsx:26]: Avoid generic naming like value

File: src/components/Stations.tsx [hunk 6]
```diff
@@ -1,45 +1,96 @@
...
-        ['Paddington', 'PAD'],
-        ['King\'s Cross', 'KGX'],
-        ['Manchester Piccadilly', 'MAN'],
-        ['Brighton', 'BTN'],
-    ]));
-
-    const [from, setFrom] = useState('LDS');
-    const [to, setTo] = useState('LDS');
+    const [allStations, setAllStations] = useState<string[]>();
+    const [allStationCodes, setAllStationCodes] = useState<Map<string, string>>();
```
Comment (WendiF) [inline src/components/Stations.tsx:10]: Likewise from previous PR - why do we have a list of stations and separately a mapping between crs and stations?

**Review by WendiF — CHANGES_REQUESTED** [review]

File: src/components/Stations.tsx (hunk 6 above)
Comment (f11xter) [reply src/components/Stations.tsx:10]: Now changed

**Pull request merged**